src/main/java/org/sagebionetworks/bridge/services/AccountService.java

```diff
@@ -0,0 +1,147 @@
+     */
+    public void deleteReauthToken(AccountId accountId) {
+        accountDao.deleteReauthToken(accountId);
+    }
+    
+    /**
+     * Create an account. If the optional consumer is passed to this method and it throws an 
+     * exception, the account will not be persisted (the consumer is executed after the persist 
+     * is executed in a transaction, however).
+     */
+    public void createAccount(Study study, Account account, Consumer<Account> afterPersistConsumer) {
+        accountDao.createAccount(study, account, afterPersistConsumer);
+    }
+    
+    /**
+     * Save account changes. Account should have been retrieved from the getAccount() method 
+     * (constructAccount() is not sufficient). If the optional consumer is passed to this method and 
+     * it throws an exception, the account will not be persisted (the consumer is executed after 
+     * the persist is executed in a transaction, however).
+     */
+    public void updateAccount(Account account, Consumer<Account> afterPersistConsumer) {
+        accountDao.updateAccount(account, afterPersistConsumer);
+    }
+    
+    /**
+     * Load, and if it exists, edit and save an account. 
+     */
+    public void editAccount(StudyIdentifier studyId, String healthCode, Consumer<Account> accountEdits) {
+        accountDao.editAccount(studyId, healthCode, accountEdits);
+    }
+    
+    /**
+     * Get an account in the context of a study by the user's ID, email address, health code,
+     * or phone number. Returns null if there is no account, it is up to callers to translate 
+     * this into the appropriate exception, if any. 
+     */
+    public Account getAccount(AccountId accountId) {
+        return accountDao.getAccount(accountId);
+    }
+    
+    /**
+     * Delete an account along with the authentication credentials.
+     */
+    public void deleteAccount(AccountId accountId) {
+        accountDao.deleteAccount(accountId);
+    }
+    
+    /**
+     * Get a page of lightweight account summaries (most importantly, the email addresses of 
+     * participants which are required for the rest of the participant APIs). 
+     * @param study
+     *      retrieve participants in this study
+     * @param search
+     *      all the parameters necessary to perform a filtered search of user account summaries, including
+     *      paging parameters.
+     */
+    public PagedResourceList<AccountSummary> getPagedAccountSummaries(Study study, AccountSummarySearch search) {
+        return accountDao.getPagedAccountSummaries(study, search);
+    }
+    
+    /**
+     * For MailChimp, and other external systems, we need a way to get a healthCode for a given email.
+     */
+    public String getHealthCodeForAccount(AccountId accountId) {
+        Account account = getAccount(accountId);
+        if (account != null) {
+            return account.getHealthCode();
+        } else {
+            return null;
+        }
+    }
```

Comment:
Is it intentional that we're re-implementing getHealthCodeForAccount() instead of calling through to the DAO?

Reply:
No, removed from the DAO.